.github/workflows/test.yaml

```diff
@@ -0,0 +1,43 @@
+name: Run Tests
+
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  setup-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Test Matrix
+        id: set-matrix
+        run: |
+          test_dirs=$(find examples/*/scripts -type f -name 'test.sh' -exec dirname {} \; | xargs dirname)
+          matrix_json=$(echo "$test_dirs" | jq -R '{"example-dir": .}' | jq -s . | jq -c .)
+          echo "matrix=$matrix_json" >> $GITHUB_OUTPUT
```

Comment:
might be cleaner to just write this in actions/github-script in javascript: https://github.com/actions/github-script